pipelines/scripts/test/util/test_synapse_workspace_manager.py

```diff
@@ -0,0 +1,258 @@
+from pipelines.scripts.util.exceptions import MaxWaitTimeNeededException
+from pipelines.scripts.util.util import Util
+import pytest
+import mock
+import requests
+import json
+import time
+import math
+
+
+SUBSCRIPTION_ID = "some_subscription"
+RESOURCE_GROUP_NAME = "some_resource_group"
+SYNAPSE_WORKSPACE_NAME = "some_synapse_workspace"
+SYNAPSE_ENDPOINT = (
+    f"https://management.azure.com/subscriptions/{SUBSCRIPTION_ID}/resourceGroups/{RESOURCE_GROUP_NAME}/providers/"
+    f"Microsoft.Synapse/workspaces/{SYNAPSE_WORKSPACE_NAME}"
+)
+
+
+def test_get_workspace_packages():
+    with mock.patch.object(requests, "get") as mock_get:
+        mock_get.return_value.json.return_value = {"value": "get_response"}
+        with mock.patch.object(SynapseWorkspaceManager, "_get_token", return_value="some_token"):
+            resp = SynapseWorkspaceManager(SYNAPSE_WORKSPACE_NAME, SUBSCRIPTION_ID, RESOURCE_GROUP_NAME).get_workspace_packages()
+            requests.get.asset_called_once_with(
+                f"{SYNAPSE_ENDPOINT}/libraries?api-version=2021-06-01",
+                headers={"Authorization": f"Bearer some_token"}
+            )
+            assert resp == "get_response"
+
+
+def test_get_workspace_packages_with_exception():
+    with mock.patch.object(requests, "get") as mock_get:
+        mock_get.return_value.json.side_effect = json.JSONDecodeError("", "", 0)
+        with mock.patch.object(SynapseWorkspaceManager, "_get_token", return_value="some_token"):
+            with pytest.raises(ValueError):
+                SynapseWorkspaceManager(SYNAPSE_WORKSPACE_NAME, SUBSCRIPTION_ID, RESOURCE_GROUP_NAME).get_workspace_packages()
+
+
+def test_upload_workspace_package():
+    package_path = "dist/some_package.whl"
+    mock_packages = [
+        {"name": "some_package.whl", "properties": {"provisioningStatus": "Succeeded"}}
+    ]
+    expected_cli_call_args = [
+        "az",
+        "synapse",
+        "workspace-package",
+        "upload",
+        "--workspace-name",
+        SYNAPSE_WORKSPACE_NAME,
+        "--package",
+        package_path,
+        "--no-progress"
+    ]
+    mock_response = {"resp": "mock_return_value"}
+    with mock.patch.object(Util, "run_az_cli_command", return_value=json.dumps(mock_response)):
+        with mock.patch.object(SynapseWorkspaceManager, "get_workspace_packages", return_value=mock_packages):
+            with mock.patch.object(time, "sleep"):
+                resp = SynapseWorkspaceManager(SYNAPSE_WORKSPACE_NAME, SUBSCRIPTION_ID, RESOURCE_GROUP_NAME).upload_workspace_package(package_path)
+                Util.run_az_cli_command.assert_called_once_with(expected_cli_call_args)
+                assert resp == mock_response
+
+
+def test_upload_workspace_package_exceeded_wait_time():
+    package_path = "dist/a_missing_package.whl"
+    mock_packages = [
+        {"name": "some_package.whl", "properties": {"provisioningStatus": "Queued"}}
+    ]
+    max_wait_time = 300  # Seconds
```

Comment:
Maybe you can define this as a class-level constants or configuration values so they're not scattered across methods